sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,416 @@
+            )
+
+        return covariance_type
+
+
+    def _fit_cluster(self, X, gm_params, seed):
+        gm_params["init_params"] = self.init_params
+        gm_params["max_iter"] = self.max_iter
+        gm_params["n_init"] = self.n_init
+        gm_params["random_state"] = seed
+
+        model = GaussianMixture(**gm_params)
+        model.fit(X)
+
+        if self.criterion == "bic":
+            criterion_value = model.bic(X)
+        else:
+            criterion_value = model.aic(X)
+
+        # change the precision of "criterion_value" based on sample size
+        criterion_value = round(criterion_value, int(np.log10(X.shape[0])))
+        results = _CollectResults(model, criterion_value, gm_params)
+        return results
+
+    def fit(self, X, y=None):
+        """Fit several Gaussian mixture models to the data.
+
+        Initialize with agglomerative clustering then
+        estimate model parameters with EM algorithm.
+        Select the best model according to the chosen
+        information criterion.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            List of n_features-dimensional data points. Each row
+            corresponds to a single data point.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        Returns
+        -------
+        self : object
+            Returns an instance of self.
+        """
+
+        covariance_type = self._check_parameters()
+        X = self._validate_data(X, dtype=[np.float64, np.float32], ensure_min_samples=1)
+
+        random_state = check_random_state(self.random_state)
+
+        # check n_components against sample size
+        n_comps = [self.max_components, self.min_components]
+        names = ["max_components", "min_components"]
+        for i in range(len(names)):
+            if n_comps[i] > X.shape[0]:
+                msg = names[i] + "must be <= n_samples, but" + names[i]
+                msg += "= {}, n_samples = {}".format(n_comps[i], X.shape[0])
+                raise ValueError(msg)
+
+        param_grid = dict(
+            covariance_type=covariance_type,
+            n_components=range(self.min_components, self.max_components + 1),
+        )
+        param_grid = list(ParameterGrid(param_grid))
+
+        seeds = random_state.randint(np.iinfo(np.int32).max, size=len(param_grid))
+
+        if parse_version(joblib.__version__) < parse_version("0.12"):
+            parallel_kwargs = {"backend": "threading"}
+        else:
+            parallel_kwargs = {"prefer": "threads"}
+
+        results = Parallel(n_jobs=self.n_jobs, verbose=self.verbose, **parallel_kwargs)(
+            delayed(self._fit_cluster)(X, gm_params, seed)
+            for gm_params, seed in zip(param_grid, seeds)
+        )
+        best_criter = [result.criterion for result in results]
+
+        if sum(best_criter == np.min(best_criter)) == 1:
```

Comment:
this all seems fine but just a suggestion - https://numpy.org/doc/stable/reference/generated/numpy.argmin.html
docs imply that for ties, argmin gives the first. so in other words if results are sorted in order of complexity, just using argmin would do what you want. (can even leave a comment to this effect, if you go this route). 

note that i think having the results sorted by complexity anyway is probably desireable?